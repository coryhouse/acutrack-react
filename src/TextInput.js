import React from "react";
import PropTypes from "prop-types";

class TextInput extends React.Component {
  render() {
    return (
      <div>
        {this.props.label && (
          <>
            <label htmlFor={this.props.id}>{this.props.label}</label>
            <br data-testid="break" />
          </>
        )}
        <input
          id={this.props.id}
          type={this.props.type}
          name={this.props.name}
          onBlur={this.props.onBlur}
          onChange={this.props.onChange}
          value={this.props.value}
          required={this.props.required}
        />
        {this.props.error && <p style={{ color: "red" }}>{this.props.error}</p>}
      </div>
    );
  }
}

TextInput.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(["email", "text", "number", "password"]),
  value: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool,
  error: PropTypes.string
};

TextInput.defaultProps = {
  required: false,
  error: ""
};

export default TextInput;
