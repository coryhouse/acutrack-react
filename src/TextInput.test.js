import React from "react";
import { render, cleanup } from "@testing-library/react";
import TextInput from "./TextInput";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("TextInput", () => {
  it("should display an error message when an error is provided on props", () => {
    // arrange
    const mockOnChange = jest.fn();

    // act
    const { getByText } = render(
      <TextInput
        error="Example error"
        label=""
        id=""
        value=""
        name=""
        onChange={mockOnChange}
      />
    );

    // assert
    getByText("Example error");
  });

  it("should contain a br tag", () => {
    // arrange

    // act
    const { getByTestId } = render(
      <TextInput label="" id="" value="" name="" onChange={jest.fn()} />
    );

    // assert
    getByTestId("break");
  });

  it("should not render an error when an error isn't passed", () => {
    const tree = renderer
      .create(
        <TextInput
          label="Example label"
          id="ID"
          value="Example value"
          name="name"
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render an error when passed an error", () => {
    const tree = renderer
      .create(
        <TextInput
          label="Example label"
          id="ID"
          error="Example error"
          value="Example value"
          name="name"
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
