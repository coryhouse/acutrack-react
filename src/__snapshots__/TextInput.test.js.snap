// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should not render an error when an error isn't passed 1`] = `
<div>
  <label
    htmlFor="ID"
  >
    Example label
  </label>
  <br
    data-testid="break"
  />
  <input
    id="ID"
    name="name"
    onChange={[Function]}
    required={false}
    value="Example value"
  />
  
</div>
`;

exports[`TextInput should render an error when passed an error 1`] = `
<div>
  <label
    htmlFor="ID"
  >
    Example label
  </label>
  <br
    data-testid="break"
  />
  <input
    id="ID"
    name="name"
    onChange={[Function]}
    required={false}
    value="Example value"
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Example error
  </p>
</div>
`;
